framework/SupervisedLearning/ARMA.py

```diff
@@ -1102,6 +1152,120 @@ def finalizeGlobalRomSegmentEvaluation(self, settings, evaluation, weights=None)
         evaluation[target] = self._transformThroughInputCDF(evaluation[target], dist, weights)
     return evaluation
 
+  ### Peak Picker ###
+  def _peakPicker(self,signal,low):
+    """
+      Peak picker
+      @ In, signal, np.array(float), signal to transform
+      @ In, low, float, required height of peaks.
+      @ Out, peaks, np.array, indices of peaks in x that satisfy all given conditions
+      @ Out, heights, np.array, boolean mask where is the residual signal
+    """
+    peaks, properties = find_peaks(signal, height=low)
+    heights = properties['peak_heights']
+    return peaks,heights
+
+  def rangeWindow(self,windowDict):
+    """
+      Collect the window index in to groups
```

Comment:
This algorithm isn't fully clear from reading the docstrings, and there's not comments in the method to guide the reader.